test/test_nn.py

```diff
@@ -9382,6 +9382,12 @@ def test_GroupNorm_empty(self, device):
             with torch.backends.cudnn.flags(enabled=False):
                 self._test_module_empty_input(mod, inp)
 
+    def test_ReflectionPad_empty(self, device):
+        for mod, inp in [
+                (torch.nn.ReflectionPad1d(2), torch.randn(0, 3, 10, device=device)),
```

Comment:
I don't think you're covering all the cases? Don't you need to check ReflectionPad1D with a 2D and 3D input with a zero batch size plus ReflectionPad2D with a 3D and 4D input with a zero batch size?

You should also add test cases that fail, where the non-batch dimension is zero and you assert you hit the appropriate error (`self.assertRaisesRegex(...)`), and a check for backward when the batch dim is zero.

Reply:
Ah yes good point. Will add them thank you.

Reply:
The checks for 2D/3D input are done in the python code in `functional.py` and the `ReflectionPad1D` does not get called unless the input is specifically `dim=3`. Same goes for 2D reflection pad (with `dim=4`). Therefore such tests are not needed.

However I have added tests for non-batch dimension being zero. The backward is tested in the `_test_module_empty_input`.